sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -490,65 +495,76 @@ export class ApplicationService extends RecordDataModelService<Application> {
-        "currentAssessment.noaApprovalStatus",
-        "offering.id",
-        "offering.offeringIntensity",
-        "offering.studyStartDate",
-        "offering.studyEndDate",
-        "offering.offeringStatus",
-        "location.id",
-        "location.name",
-        "pirDeniedReasonId.id",
-        "pirDeniedReasonId.reason",
-        "programYear.id",
-        "programYear.formName",
-        "programYear.startDate",
-        "programYear.endDate",
-        "applicationException.exceptionStatus",
-        "application.submittedDate",
-      ])
-      .leftJoin("application.currentAssessment", "currentAssessment")
-      .leftJoin("currentAssessment.offering", "offering")
-      .leftJoin("application.location", "location")
-      .leftJoin("location.institution", "institution")
-      .leftJoin("institution.institutionType", "institutionType")
-      .innerJoin("application.student", "student")
-      .innerJoin("student.user", "user")
-      .innerJoin("application.programYear", "programYear")
-      .leftJoin("application.pirDeniedReasonId", "pirDeniedReasonId")
-      .leftJoin("application.applicationException", "applicationException")
-      .where("application.id = :applicationIdParam", {
-        applicationIdParam: applicationId,
-      })
-      .andWhere("application.applicationStatus != :overwrittenStatus", {
-        overwrittenStatus: ApplicationStatus.overwritten,
-      });
-    if (userId) {
-      applicationQuery.andWhere("user.id = :userId", { userId });
-    }
-    return applicationQuery.getOne();
+    return this.repo.findOne({
+      select: {
+        id: true,
+        data: !!options?.loadDynamicData as unknown,
+        applicationStatus: true,
+        pirStatus: true,
+        applicationStatusUpdatedOn: true,
+        applicationNumber: true,
+        pirDeniedOtherDesc: true,
+        submittedDate: true,
+        applicationException: {
+          id: true,
+          exceptionStatus: true,
+        },
+        currentAssessment: {
+          id: true,
+          assessmentWorkflowId: true,
+          noaApprovalStatus: true,
+          offering: {
+            id: true,
+            offeringIntensity: true,
+            studyStartDate: true,
+            studyEndDate: true,
+            offeringStatus: true,
+          },
+        },
+        location: {
+          id: true,
+          name: true,
+        },
+        pirDeniedReasonId: {
+          id: true,
+          reason: true,
+        },
+        programYear: {
+          id: true,
+          formName: true,
+          startDate: true,
+          endDate: true,
+        },
+      },
+      relations: {
+        applicationException: true,
+        currentAssessment: { offering: true },
+        location: true,
+        pirDeniedReasonId: true,
+        programYear: true,
+      },
+      where: {
+        id: applicationId,
+        applicationStatus: Not(ApplicationStatus.overwritten),
+        student: {
+          id: options?.studentId,
```

Comment:
The options and studentId are optional parameters. Is that a valid scenario, if we do no have options or studentId, the value will be undefined. Dont we need to change the query accordingly, if null is passed?